Orange/widgets/widget.py

```diff
@@ -508,6 +508,14 @@ def closeContext(self):
         reinitializing the user interface (e.g. combo boxes) with the new
         data.
         """
+        # Clear schema-only settings when *closing* the context
+        # FIXME: Sadly, schema-only settings aren't cleared when non-contextual
```

Comment:
Actually, they are.

As non-context-aware widgets only load settings when a new instance of the widget is put on the canvas, this is achieved by not updating default values. See https://github.com/biolab/orange3/blob/master/Orange/widgets/settings.py#L508